lib/errors.ts

```diff
@@ -207,9 +212,13 @@ export async function handleError(error: Error) {
 		if (!process.env.BALENARC_NO_SENTRY) {
 			await sentryCaptureException(error);
 		}
-
-		// Unhandled/unexpected error: ensure that the process terminates.
+	}
+	if (error instanceof SIGINTError || !isExpectedError) {
+		// SIGINT or unexpected error: ensure that the process terminates.
 		// The exit error code was set above through `process.exitCode`.
+		if (process.env.DEBUG) {
+			console.error('[debug] Forceful process exit');
+		}
 		process.exit();
```

Comment:
Calling `process.exit()` here in the event of SIGINT is necessary (arguably) when the `awaitInterruptibleTask` function is used (see `helpers.ts` in this PR). Without `process.exit()` here, the CLI process would stay running until the "other" (non-sigint) promise in the `awaitInterruptibleTask` function is resolved, which substantially defeats the purpose of SIGINT. (Without `process.exit()`, the process would exit gracefully after that other promise resolved, but it could take a while if the promise is for a long-running build...)
The important / nice thing though, is that `process.exit()` is only called at this very late stage, after the promise rejection bubbled up all the way to centralized error handling in `errors.ts`, and even got reported to Sentry.io if it was an "unexpected error" that happened after SIGINT.